ui/components/multichain/app-header/app-header.js

```diff
@@ -0,0 +1,213 @@
+              )}
+
+              <AccountPicker
+                address={identity.address}
+                name={identity.name}
+                onClick={() => dispatch(toggleAccountMenu())}
+              />
+              <Box
+                display={DISPLAY.FLEX}
+                alignItems={AlignItems.center}
+                justifyContent={JustifyContent.spaceBetween}
+              >
+                {showStatus ? (
+                  <ConnectedStatusIndicator
+                    onClick={() => history.push(CONNECTED_ACCOUNTS_ROUTE)}
+                  />
+                ) : null}
+                <Box
+                  ref={menuRef}
+                  display={DISPLAY.FLEX}
+                  justifyContent={JustifyContent.flexEnd}
+                  width={BLOCK_SIZES.FULL}
+                >
+                  <ButtonIcon
+                    iconName={ICON_NAMES.MORE_VERTICAL}
+                    data-testid="account-options-menu-button"
+                    ariaLabel="NEEDS NEW TRANSLATED LABEL"
+                    onClick={() => {
+                      trackEvent({
+                        event: MetaMetricsEventName.NavAccountMenuOpened,
+                        category: MetaMetricsEventCategory.Navigation,
+                        properties: {
+                          location: 'Home',
+                        },
+                      });
+                      setAccountOptionsMenuOpen(true);
+                    }}
+                  />
+                </Box>
+              </Box>
+              {accountOptionsMenuOpen ? (
+                <GlobalMenu
+                  anchorElement={menuRef.current}
+                  closeMenu={() => setAccountOptionsMenuOpen(false)}
+                />
+              ) : null}
+            </Box>
+          ) : (
+            <Box
+              display={DISPLAY.FLEX}
+              className={`multichain-app-header__lock-contents ${
+                isUnlocked && !popupStatus ? 'multichain-app-header-shadow' : ''
+              }`}
+              alignItems={AlignItems.center}
+              width={BLOCK_SIZES.FULL}
+              justifyContent={JustifyContent.spaceBetween}
+              backgroundColor={BackgroundColor.backgroundDefault}
+              padding={2}
+              gap={2}
+            >
+              <PickerNetwork
+                label={currentNetwork?.nickname}
+                src={currentNetwork?.rpcPrefs?.imageUrl}
+                onClick={() => dispatch(toggleNetworkMenu())}
+              />
+              <MetafoxLogo
+                unsetIconHeight
+                onClick={async () => {
+                  if (onClick) {
+                    await onClick();
+                  }
+                  history.push(DEFAULT_ROUTE);
+                }}
+              />
+            </Box>
+          )}
+        </>
+      </Box>
+    </>
+  );
+};
+
+AppHeader.propTypes = {
+  /**
+   * The onClick handler to be passed to the App Header component
```

Comment:
Maybe we can be more specific here say that it's passed to the MetaMask logo?
```suggestion
   * The onClick handler to be passed to the MetaMask Logo in the App Header
```